### DISCUSSION TITLE
Running db backup

### DISCUSSION THREAD

**Discussion by steakhutzeee** [firefly-iii/firefly-iii#4702] [Firefly III]
Hi :)

Reading the documentation I see that to backup the db I have to backup the db docker volume. 

Now I would like to ask if it is ok to backup content of db volume while the db is running?

Actually i added an ad hoc image to my compose to automatically backup the db but was curious to ask if this is really needed.

Thanks!

**Comment by JC5** [comment]
What I do myself is shut down the DB just in case. I don't know if it'll hurt.

**Reply by steakhutzeee** [reply]
> 
> 
> What I do myself is shut down the DB just in case. I don't know if it'll hurt.

I think that is the correct way to backup it "physically".

Found here a good description [https://dev.mysql.com/doc/refman/8.0/en/backup-types.html](https://dev.mysql.com/doc/refman/8.0/en/backup-types.html)

So i think it's better to have both physical and logical backup. So for the same architecture/hardware you can just restore your saved docker volume. In case of restore on a complete different machine the restore can be performed from the logical backup.

**Comment by Genygo** [comment]
For a logical backup you could have a look at #4270
If you haven't already figured out this part.

**Reply by steakhutzeee** [reply]
Yep, I have implemented the container suggested there :D
